### DISCUSSION TITLE
Example to (de)serialize nested optional fields from the JSON string and to JSON

### DISCUSSION THREAD

**Discussion by matu3ba** [nlohmann/json#3560] [Q&A] [answered]
The showed examples did not hint onto any such functionality, so I am curious if this is a supported use case.
Ideally, the existence of the optional field would be written to a field inside the whole struct that defines the json.

If it is: Can you tell me, if and how much code bloat to expect?

NOTE: I searched for "optional fields" and "optional JSON fields" without, but found no related discussion.

**Comment by nlohmann** [comment, marked as answer]
Can you please provide an example what you want to achieve?

**Reply by matu3ba** [reply]
Assume a JSON schema
```json
{
  "required_field":"data1","optional_field":{"req_field_of_optional":"data2","nested_opt_field":"data3"}
}
```
which fills
```c++
struct json_repr {
  typeof(data1) required_field;
  bool optional_field_isUsed;
  typeof(data2) req_field_of_optional;
  bool nested_opt_field_isUsed;
  typeof(data3) nested_opt_field;
};
```
Another conforming JSON schema would be
```json
{
  "required_field":"data1"
}
```
and another one
```json
{
  "required_field":"data1","optional_field":{"req_field_of_optional":"data2"}
}
```

**Reply by nlohmann** [reply]
So is this about how a `to_json`/`from_json` function for such a struct should look like?

**Reply by matu3ba** [reply]
Yes.

**Reply by nlohmann** [reply]
- For `from_json`, you can use [contains](https://json.nlohmann.me/api/basic_json/contains/) to check whether the optional keys are present and act accordingly.
- For `to_json`, you would need to decide based on the value of optional fields whether or not to serialize them by adding the respective keys to the JSON value.

**Reply by nlohmann** [reply]
@matu3ba Do you need further assistance with this issue?

**Reply by matu3ba** [reply]
No. I do consider this as answered. Thanks alot for the response.
